kubernetes/utils/create_from_yaml.py

```diff
@@ -60,46 +66,77 @@ def create_from_yaml(
         processing of the request.
         Valid values are: - All: all dry run stages will be processed
     """
+    if path.exists(yaml_file):
+        with open(path.abspath(yaml_file)) as f:
+            content = f.read()
+            try:
+                yaml_file = StringIO(content)
+            except TypeError:
+                yaml_file = StringIO(content.decode('utf-8'))
+
+    yml_document_all = yaml.safe_load_all(yaml_file)
+    # Load all documents from a single YAML file
+    fail_exceptions = []
+
+    for yml_document in yml_document_all:
+        exceptions = create_from_dict(k8s_client, yml_document, verbose,
+                                      **kwargs)
+        if exceptions:
+            fail_exceptions.extend(exceptions)
+
+    if fail_exceptions:
+        raise FailToCreateError(fail_exceptions)
+
+
+def create_from_dict(k8s_client, data, verbose=False, **kwargs):
+    """
+    Perform an action from a dictionary containing one or more valid kubernetes
```

Comment:
"one or more" is not really correct is it?
it only takes one kubernetes object, that one object can be a kubernetes-list but it still one object you are inputting
e.g. you cannot input two different types of objects, like a service and a deployment.

if it would take more than one object data would need to be a list/sequence of dicts/mappings

Reply:
Yes, you are correct. The dictionary itself contains one item, this item by itself can contain more objects. See my last commit.